### ISSUE TITLE
Allow user to create a production order 

### ISSUE THREAD

**Issue by jmiranda** [openboxes/openboxes#146]
Inputs: 
- Product to be manufactured
- Raw materials to be process (see #144 )
- Location where manufacturing/processing will take place
- Date ordered
- Date promised 
- Additional work to be performed (optional?)

Basic Workflow:
- Choose a bill of materials (BoM) to be manufactured.
- Verify that there's enough QoH for the items in the BoM.
- Transfer quantities of items from bill of materials to the order.
- Reserve all items in the bill of material so that they are not available to be picked for other orders.
- Once the production order is complete, the items will be issued and the BoM quantity for the BoM item will be incremented accordingly.

Should probably handle backflushing by default so we don't issue stock to the work order until the process is completed in the system. Happy to discuss

**Comment by kigsmtua** [comment]
I would like a go at this

**Comment by jmiranda** [comment]
Ok sounds good. Let me know if you need any help getting started. I'm still not exactly sure about the UI and workflow but we can discuss that here and maybe add some wireframes.

**Comment by XXyomerengues** [comment]
I believe it should work as a shipment, date ordered & date promised are like shipping & arrival; the picking is also already programmed, destination would be the new product increased at the inventory, all it needs to be added is the protocol for the production: I need to make 25 "x" products, each needs 1a, 1b & 2c raw materials, then "x" is the destination; 25a,25b & 50c are added automatically to the "shipment" if available, if not, the same warning appears: not enough materials. What do you think?

**Comment by jmiranda** [comment]
Yeah that makes sense. I've seen a few systems that add workflows that include other resources (services, workstations, etc) but I'm guessing that might be overkill to start.

**Comment by jmiranda** [comment]
@XXyomerengues by the way, are you commenting on this ticket because you actually need this feature?

**Comment by XXyomerengues** [comment]
No, I just wanted to provide the idea :-)
